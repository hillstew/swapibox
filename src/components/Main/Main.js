import React, { Component } from "react";
// import Header from "../Header/Header"
import { Controls } from "../Controls/Controls";
import { CardContainer } from "../CardContainer/CardContainer";
import { getCleanVehicles, fetchData, getCleanPeople, getCleanPlanets} from "../../Helper/Helper";

class Main extends Component {
  constructor() {
    super();
    this.state = {
      view: "",
      planets: [],
      vehicles: [],
      people: [],
      favorites: []
    };
  }

  handleChange = async event => {
    const { name: view } = event.target;
    if (this.state[view].length === 0) {
      let dataArray = [];
      const data = await fetchData(`https://swapi.co/api/${view}/`);
      dataArray.push(...data.results)
      var cleanedData = await this.handleData(view, dataArray);
    }
    this.setState({ view, [view]: cleanedData });
  };

  handleData = (view, data) => {
    switch (view) {
      case "planets":
        return getCleanPlanets(data);
      case "people":
        return getCleanPeople(data);
      case "vehicles":
        return getCleanVehicles(data);
      default:
        return;
    }
  };

  render() {
    const { view, planets, vehicles, people, favorites } = this.state;
    return (
      <div className="main-div">
        {view === "" && <Controls handleChange={this.handleChange} />}
        {view !== "" && (
          <CardContainer
            view={view}
            planets={planets}
            vehicles={vehicles}
            people={people}
          />
        )}
      </div>
    );
  }
}

export default Main;
