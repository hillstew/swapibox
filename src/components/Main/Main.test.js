import React from "react";
import Main from "../Main/Main";
import { shallow } from "enzyme";

describe("Main", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
