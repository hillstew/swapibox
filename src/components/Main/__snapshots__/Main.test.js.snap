// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
<div
  className="main-div"
>
  <Controls
    handleChange={[Function]}
  />
</div>
`;
